TicTacToeClassLibrary/TicTacToe.cs

```diff
@@ -46,19 +41,30 @@ public void WriteSign(int position)
             }
         }
 
-        public Board Create()
+        public Board CreateBoard()
         {
-            Console.WriteLine("Please select a board type (3x3, 4x4)");
+            Console.WriteLine("Please select a board type (3x3, 4x4):");
             string boardType = Console.ReadLine();
-            Board board = _boardFactory.Create(boardType);
-            _mementos = new List<IMemento>();
-            return board;
```

Comment:
There is no creation of a new _mementos list (or deletion of all its elements). So when a player presses "u" when starting a new game, the board from the previous game is returned - this is a bug.